src/EnergyPlus/ZoneTempPredictorCorrector.cc

```diff
@@ -4760,22 +4737,14 @@ void CalcPredictedHumidityRatio(EnergyPlusData &state, int const ZoneNum, Real64
         // are currently set to zero when the CTF only version is used.
 
         // if no surface in the zone uses EMPD or HAMT then zero
-        bool no_ht_EMPD_or_HAMT(true);
-        for (int i = Zone(ZoneNum).HTSurfaceFirst, e = Zone(ZoneNum).HTSurfaceLast; i <= e; ++i) {
-            auto const &htAlgo(state.dataSurface->Surface(i).HeatTransferAlgorithm);
-            if ((htAlgo == DataSurfaces::HeatTransferModel::EMPD) || (htAlgo == DataSurfaces::HeatTransferModel::HAMT)) {
-                no_ht_EMPD_or_HAMT = false;
-                break;
-            }
-        }
-        if (no_ht_EMPD_or_HAMT) {
+        if (state.dataHeatBal->Zone(ZoneNum).noEMPDorHAMT) {
             state.dataHeatBalFanSys->SumHmARaW(ZoneNum) = 0.0;
             state.dataHeatBalFanSys->SumHmARa(ZoneNum) = 0.0;
         }
```

Comment:
Searching the code for SumHmARaW, I don't think these resets should be here, or at line 5637. I think these are left over from way back before EMPD and HAMT were added.  I'm going to check the archive to be sure. There's another place [here in `CalcHeatBalanceAir`](https://github.com/NREL/EnergyPlus/blob/fb5dce86f0a63aa46f11c9ddacdd944e5f247bd5/src/EnergyPlus/HeatBalanceAirManager.cc#L4782-L4785) that's also resetting these to zero with no check for EMPD or HAMT.  If these are appropriately place, then maybe we ditch the zone level flags and put if(AnyEMPD || AnyHAMT) around this block and dump the zone-level flags. And rewrite these bulk array assignments as loops.

Reply:
And the reset to zero in `CalcHeatBalanceAir` shouldn't be there either. It's from way back when and there used to be rests of SumHATSurf  = 0.0 there as well.
The only place now that `SumHmARa` and `SumHmARaW` are accumulated is [here in `CalcHeatBalanceInsideSurf2`](https://github.com/NREL/EnergyPlus/blob/fb5dce86f0a63aa46f11c9ddacdd944e5f247bd5/src/EnergyPlus/HeatBalanceSurfaceManager.cc#L7600-L7601). I would delete the zone-level `noEMPTorHAMT` flage and just reset all three of these before the surface loop:
```
        state.dataHeatBalFanSys->SumHmAW(zoneNum) = 0.0;
        state.dataHeatBalFanSys->SumHmARa(zoneNum) = 0.0;
        state.dataHeatBalFanSys->SumHmARaW(zoneNum) = 0.0;
```

Reply:
OK, I have 0 diffs on HAMT*.idfs and a few other files.